File: 1/Hardware.java
class Hardware extends Product
{
	private String brand;
	private String model;

	// since java doesn't support optional parameter, overloading the 
	// constructor was done instead

	// construct also with current class parent's constructor
	public Hardware(String brand, String model, int price, String idProduct)
	{
		super(price, idProduct);
		this.setBrand(brand);
		this.setModel(model);
	}

	// construct for this class only
	public Hardware(String brand, String model)
	{
		this.setBrand(brand);
		this.setModel(model);
	}

	// default construct
	public Hardware(){}

	public String getBrand()
	{
		return brand;
	}

	public void setBrand(String brand)
	{
		this.brand = brand;
	}

	public String getModel()
	{
		return model;
	}

	public void setModel(String model)
	{
		this.model = model;
	}

	public void printHardware()
	{
		System.out.println("Brand : " + this.brand);
		System.out.println("Model : " + this.model);
	}
}

File: 2/Hardware.py
from Product import Product

class Hardware(Product):
	__brand: str
	__model: str

	# construct with optional args
	def __init__(self, brand: str, model: str, price=None, idProduct=None):
		super().__init__(price, idProduct)

		self.__brand = brand
		self.__model = model
	
	def setBrand(self, brand):
		self.__brand = brand

	def getBrand(self):
		return self.__brand

	def setModel(self, model):
		self.__model = model

	def getModel(self):
		return self.__model

	def printHardware(self):
		print("Brand : ", self.__brand)
		print("Model : ", self.__model)